pkg/logql/lex.go

```diff
@@ -240,21 +240,43 @@ func tryScanDuration(number string, l *scanner.Scanner) (time.Duration, bool) {
 		return 0, false
 	}
 	// we've found more characters before a whitespace or the end
-	d, err := time.ParseDuration(sb.String())
+	durationString := sb.String()
+	duration, err := parseDuration(durationString)
 	if err != nil {
 		return 0, false
 	}
+
 	// we need to consume the scanner, now that we know this is a duration.
 	for i := 0; i < consumed; i++ {
 		_ = l.Next()
 	}
-	return d, true
+
+	return duration, true
+}
+
+func parseDuration(d string) (time.Duration, error) {
+	var duration time.Duration
+	// Try to parse promql style durations first, to ensure that we support the same duration
+	// units as promql
+	prometheusDuration, err := model.ParseDuration(d)
+	if err != nil {
+		// Fall back to standard library's time.ParseDuration if a promql style
+		// duration couldn't be parsed.
+		duration, err = time.ParseDuration(d)
+		if err != nil {
+			return 0, err
+		}
+	} else {
+		duration = time.Duration(prometheusDuration)
+	}
+
+	return duration, nil
 }
 
 func isDurationRune(r rune) bool {
 	// "ns", "us" (or "µs"), "ms", "s", "m", "h".
```

Comment:
I think we should keep this comment in sync with what we have in the switch case. While we are at it, can you please also order it the same manner as the switch case?

Reply:
Thanks for the feedback! Agreed, and done.